src/main.cpp

```diff
@@ -624,11 +592,29 @@ int main(int argc, char *argv[])
         return EXIT_FAILURE;
     }
 
+    // NOTE: Even if the singleton objects do not exist when this function is called, invoking the
```

Comment:
I moved the contents of the clean up callback function here.

```
SDL_GL_DeleteContext(glcontext);
```

was being called before

```
VideoEngine::SingletonDestroy();
```

This caused the OpenGL context to be destroyed before the clean up its resources actually happened.  So, now the VideoEngine is cleaned up prior to deleting the OpenGL context.